src/odemis/gui/cont/tabs/fastem_project_settings_tab.py

```diff
@@ -255,7 +260,45 @@ def on_evt_hfw_sb_ctrl(self, evt):
         if not ctrl:
             return
 
-        value = ctrl.GetClientData(ctrl.GetSelection())
+        ctrl_value = ctrl.GetValue()
+        i = ctrl.GetSelection()
+        # EVT_COMBOBOX is triggered when the user selects an item from the dropdown
+        if i != wx.NOT_FOUND and ctrl.Items[i] == ctrl_value:
```

Comment:
Can you add a comment explaining how this `if` conditions are linked to the "EVT_COMBOBOX"? Wouldn't it be simpler to check the event type (using `isinstance`), or even having each event bound to a different callback? 
Or does this `if` actually handle special corner cases which make one event having to be handled by the other event type handler? (eg, the user types in an existing value).

Reply:
I just followed a similar approach to https://github.com/delmic/odemis/blob/c16892410729f7078b3e0403a480852aa20d54b8/src/odemis/gui/conf/util.py#L880

You are right, I can separate the 2 event callbacks.